src/registry.rs

```diff
@@ -11,8 +11,32 @@ use fs_err as fs;
 
 use super::{iso_3166, iso_639_2, iso_639_3, Error, Language, Locale, Territory};
 
-/// All ISO codes are expected to live in this location
-const ISO_CODES_BASE: &str = "/usr/share/iso-codes/json";
+fn find_iso_codes_base() -> Result<String, Error> {
+    use std::env;
+
+    let candidates = env::var("XDG_DATA_DIRS")
+        .ok()
+        .map(|dirs| {
+            dirs.split(':')
+                .map(|dir| format!("{dir}/iso-codes/json"))
+                .collect::<Vec<_>>()
+        })
+        .unwrap_or_else(|| vec![
+            "/usr/local/share/iso-codes/json".into(),
+            "/usr/share/iso-codes/json".into(),
+        ]);
+
+    for path in candidates {
+        if fs::metadata(format!("{}/iso_3166-1.json", &path)).is_ok() {
+            return Ok(path);
+        }
+    }
+
+    Err(Error::IO(std::io::Error::new(
+        std::io::ErrorKind::NotFound,
+        "Could not find iso-codes json data in XDG_DATA_DIRS or standard paths",
```

Comment:
Perhaps being very explicit would be nice for packagers/implementers?

```suggestion
        "No iso-codes json data found in XDG_DATA_DIRS, /usr/share/iso-codes/json or /usr/local/share/iso-codes/json",
```

Reply:
Sounds good to me 👍